libp2p/NodeTable.cpp

```diff
@@ -233,25 +237,27 @@ void NodeTable::doDiscover(NodeID _node, unsigned _round, shared_ptr<set<shared_
         return;
     }
 
-    m_timers.schedule(c_reqTimeout.count() * 2, [this, _node, _round, _tried](boost::system::error_code const& _ec)
-    {
-        if (_ec)
-            // we can't use m_logger here, because captured this might be already destroyed
-            clog(VerbosityDebug, "discov")
-                << "Discovery timer was probably cancelled: " << _ec.value() << " "
-                << _ec.message();
-
-        if (_ec.value() == boost::asio::error::operation_aborted || m_timers.isStopped())
-            return;
-
-        // error::operation_aborted means that the timer was probably aborted.
-        // It usually happens when "this" object is deallocated, in which case
-        // subsequent call to doDiscover() would cause a crash. We can not rely on
-        // m_timers.isStopped(), because "this" pointer was captured by the lambda,
-        // and therefore, in case of deallocation m_timers object no longer exists.
+    m_discoveryTimer->expires_from_now(c_discoveryRoundIntervalMs);
+    auto discoveryTimer{m_discoveryTimer};
+    m_discoveryTimer->async_wait(
+        [this, discoveryTimer, _node, _round, _tried](boost::system::error_code const& _ec) {
+            // We can't use m_logger here if there's an error because captured this might already be
```

Comment:
here too, move comment

Reply:
So I purposely put this comment above the "if" case since it applies to both the "if" and "else if" and I didn't want to include the same comment in both cases.